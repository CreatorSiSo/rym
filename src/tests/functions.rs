use crate::{expr_parser, insta_assert_parser, item_parser};

#[test]
fn empty() {
	insta_assert_parser! {
		item_parser(expr_parser());

		"func empty() {}",
		"func empty();",
		// "func empty() = Unit",
	}
}

#[test]
fn params() {
	insta_assert_parser! {
		item_parser(expr_parser());

		"func params(one, two, three);",
		"func params(self, other);",
		// "func add(mut self, other);",
		// "func combine((l_start, l_end): (Usize, Usize), (r_start, r_end): (Usize, Usize));",
		// "func combine(Record { inner }: Record, .{ inner }: Record);",
	}
}
