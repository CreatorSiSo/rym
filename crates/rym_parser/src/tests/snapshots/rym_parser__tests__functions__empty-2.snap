---
source: src/tests/functions.rs
expression: "&snapshot"
---
--- Input ---
func empty();
---

Some(
    Spanned(
        Func {
            name: Spanned(
                "empty",
                5..10,
            ),
            params: [],
            rhs: None,
        },
        0..13,
    ),
)

--- Errors ---
[]
---
